Seq2SeqSharp/Tools/BaseSeq2SeqFramework.cs

```diff
@@ -388,6 +388,9 @@ public void Train(int maxTrainingEpoch, ICorpus<IPairBatch> trainCorpus, ICorpus
                 // Train one epoch over given devices. Forward part is implemented in RunForwardOnSingleDevice function in below, 
                 // backward, weights updates and other parts are implemented in the framework. You can see them in BaseSeq2SeqFramework.cs
                 TrainOneEpoch(i, trainCorpus, validCorpusList, learningRate, optimizer, taskId2metrics, decodingOptions, RunForwardOnSingleDevice);
+
+                // send progress reporting in the form of a percentage value (0-100%)
+                Logger.WriteLine(Logger.Level.info, "", (int)(100 * (i + 1) / maxTrainingEpoch));
```

Comment:
Besides the progress value, we could also show prompt likes "Finished Epoch: xxx%"

Reply:
Let us start with this simple demo and then we can add more options later.

Reply:
Good idea, but that is not possible here because then the progress value will be interpreted as formatting (`params object[] args` is used for formatting the message). That is the reason for why the message must be empty for progress reporting. We should explain this in the code or find a better solution later...

Reply:
Maybe we could create a separated method for progress reporting? such as Logger.ReportProgress(....) ? And it could report progress to listeners delegated by callback handlers ?